.github/actions/unit-tests/action.yml

```diff
@@ -18,13 +18,14 @@ runs:
     - name: Install researcher dependencies
       run: | 
         source ~/.bashrc
+        conda clean -i -y
         ./scripts/configure_conda researcher
         ./scripts/fedbiomed_run network
       shell: bash
 
     - name: Run unit tests 
       run: |
-          PYTHONPATH=${PYTHONPATH:-$PWD} conda run -n fedbiomed-researcher nosetests -w ${{ inputs.test-dir }} --cover-xml --cover-erase --with-coverage --cover-package=fedbiomed --with-xunit -v --process-restartworker
+          PYTHONPATH=${PYTHONPATH:-$PWD} conda run -n fedbiomed-researcher pytest -v --cov=fedbiomed --cov-report term --cov-report xml:${{ inputs.test-dir }}/coverage.xml ${{ inputs.test-dir }}
```

Comment:
Hi @mvesin, 

Thank you for the great effort! 

I got an error while running unittest tests with pytest. I think it is due to my environment but others may have the same error too. We may the change the way that we test `test_environ_01_initialize_common_variables_01`. 

```
tests/test_environ_common.py::TestEnviron::test_environ_01_initialize_common_variables_01 - AssertionError: '/user/scansiz/home/projects/fedbiomed-dev/fedbiomed' != '/home/scansiz/projects/fedbiomed-dev/fedbiomed'
```

Reply:
Thanks @srcansiz , I'm going to robustify this test.

Nevertheless, I suspect it might be due to the command you launch for test, not the migration to pytest.
Documented command in `tests/README.md` is:

```
cd tests
pytest -v --cov=fedbiomed --cov-report term --cov-report xml:coverage.xml
```

... which works fine for me, but if I run from `/home/mvesin/GIT/fedbiomed/fedbiomed/tests` the following then I get (and find it quite normal) the same error that you reported:

```
# use an alternate path to reach the dir
$ pytest -v --cov=fedbiomed --cov-report term --cov-report xml:coverage.xml /user/mvesin/home/mvesin/GIT/fedbiomed/fedbiomed/tests
...
FAILED ../../../../../../user/mvesin/home/GIT/fedbiomed/fedbiomed/tests/test_environ_common.py::TestEnviron::test_environ_01_initialize_common_variables_01 - AssertionError: '/home/mvesin/GIT/fedbiomed/fedbiomed' != '/user/mvesin/hom...

```

Reply:
Thank you @mvesin, 

I also tested to run `pytest` in the `tests`  directory. It still fails on the same test.

Reply:
@srcansiz : pushed a patch for this, can you confirm that it also fixes the issue for you ?

Reply:
Hi @mvesin,

Yes it worked! thank you